sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -782,7 +782,11 @@ def _intervals_to_df(
         [
             _interval_to_df(
                 snapshot,
-                *snapshot.inclusive_exclusive(start, end),
+                *(
+                    snapshot.get_remove_interval(start, end)
+                    if is_removed
+                    else snapshot.inclusive_exclusive(start, end)
```

Comment:
Can we somehow name this or `get_remove_interval` to highlight their relation? Eg. `get_add_interval` / `get_remove_interval`. Or `inclusive_exclusive` and `inclusive_exclusive_for_remove`. Or maybe 
```
def inclusive_exclusive(start, end, for_remove: bool = False)
```
Do we even need a separate method btw?

Reply:
I went with the `for_remove` property. Let me know what you think: https://github.com/TobikoData/sqlmesh/pull/921/commits/3cb6541f9044bdf45e6430295f5c365ec6544572